src/nettests/index.js

```diff
@@ -1,9 +1,106 @@
+import camelCase from 'camelcase'
+
+import percentage from '../cli/output/percentage'
+import wait from '../cli/output/wait'
+
+import {
+  Result,
+  Measurement
+} from '../config/db'
+
 import HTTPHeaderFieldManipulation from './http-header-field-manipulation'
 import HTTPInvalidRequestLine from './http-invalid-request-line'
 import WebConnectivity from './web-connectivity'
 import NDT from './ndt'
 import Base from './base'
 
+export const makeOoni = () => {
+  let dbOperations = [],
+      progress = null,
+      reportId = null,
+      uploaded = false,
+      localReportId = null,
+      isMk = false
+
+  let measurement = Measurement.build({
+    state: 'active'
+  })
+
+  const init = (nt) => {
+    // XXX we use duck typing to see if the argument is a measurement Kit
+    // nettest. Maybe we should change this.
+    if (nt.test && nt.name && nt.setOptions) isMk = true
+
+    measurement.name = camelCase(nt.name)
+    measurement.reportFile = measurement.makeReportFile()
+    dbOperations.push(measurement.save())
+
+    if (isMk) {
+      nt.test.set_options('no_file_report', '0')
+      nt.test.set_output_filepath(measurement.reportFile)
+      nt.on('log', (severity, message) => {
+        // XXX this a workaround due to a bug in MK
+        // I actually also need to know if the report has been created
+        if (message.startsWith('Report ID:') && reportId === null) {
+          reportId = message.split(':')[1].trim()
+          uploaded = true
+        }
+      })
+      nt.on('entry', entry => {
```

Comment:
I am assuming that `on` here in node has the property that you can register more than one callback for event; does this make sense, or is the assumption wrong? (Should MK do the same? In theory it could.)

Reply:
yes correct.

Reply:
Okay, see https://github.com/measurement-kit/measurement-kit/issues/1500